src/v/config/bounded_property.h

```diff
@@ -248,13 +263,13 @@ class bounded_property : public property<T> {
         if constexpr (reflection::is_std_optional<outer_type>) {
             if (val.has_value()) {
                 return property<T>::update_value(
-                  std::move(_bounds.clamp(val.value())));
+                  std::move(clamp_with_bounds(val.value())));
```

Comment:
nit: looks like std::move is unnecessary since update_value is taking a temporary?